### PR TITLE
[FIX,TST] Adds unit tests for ma_pca functions and fixes a bug in ma_pca

### LINKED ISSUE #522: Add rigorous unit testing to PCA module
### Additional Detail
We need to add unit tests to the PCA module.

### Next Steps
Even though the methods are "private" the unit tests can still call them easily since there is no real "private" status in Python. We can simply write the unit tests as calling those methods.

### PR DESCRIPTION
**Pull request by eurunuela**
Closes #522 .

Changes proposed in this pull request:

- Adds unit tests to almost every function in `ma_pca.py`.
- Fixes bug in `ma_pca.py`.

### CODE REVIEW

File: tedana/tests/test_mapca.py [hunk 1]
```diff
@@ -6,6 +6,50 @@
...
+    test_order = 0
+    ord_out, w, trivwin = _check_order(test_order)
+    assert ord_out == test_order
+    assert trivwin
+
+    test_order = 1
+    ord_out, w, trivwin = _check_order(test_order)
+    assert ord_out == test_order
+    assert w == 1
+    assert trivwin
```
Comment (emdupre) [inline tedana/tests/test_mapca.py]: Any way to separately test when `trivwin` is `False` ?

Comment (eurunuela) [reply tedana/tests/test_mapca.py]: Must have missed that one. Just added this. Thanks for catch @emdupre! 💯

File: tedana/tests/test_mapca.py [hunk 2]
```diff
@@ -6,6 +6,55 @@
 import nibabel as nib
 from tedana import decomposition
 from pytest import raises
+from tedana.decomposition.ma_pca import _autocorr, _check_order, _parzen_win
+from tedana.decomposition.ma_pca import _subsampling, _kurtn, _icatb_svd, _eigensp_adj
+
+
+def test_autocorr():
+    """
+    Uni test on _autocorr function
```
Comment (tsalo) [inline tedana/tests/test_mapca.py]:
```suggestion
    Unit test on _autocorr function
```

**Comment by eurunuela** [comment]
Thanks for the feedback @tsalo!

**Comment by eurunuela** [comment]
That’s right. I checked everything on a Python session.

**Pull request merged**